### PR TITLE
[fix] pnext pointer from the LevelZero descriptor types

### PR DESCRIPTION
**Pull request by stratika**
This PR is related to the issue [555](https://github.com/beehive-lab/TornadoVM/issues/555) from TornadoVM, and #9.

The problem was that we use in the JNI part the LevelZero descriptors (e.g., [ze_event_pool_desc_t](https://spec.oneapi.io/level-zero/latest/core/api.html#structze__event__pool__desc__t) descriptor), which contain a `pNext` pointer. 

In the Java world this pointer is a long value that should hold the address that it will point to. However, if an address is not set the default value is `-1`.

On the other hand, the `LevelZero` driver is expecting to see `null` if the address is not set. And this was causing the following segfault.
Screenshot from 2024-09-16 15-35-04

**This PR contains the following changes:**
- I removed from some descriptor classes the initialization of the `pNext` variable, since this was initialized with the default value in the super class.
- I added a new test (TestZelloWorld.java) that follows the [example](https://github.com/oneapi-src/level-zero/blob/master/samples/zello_world/zello_world.cpp) from the level-zero repository.
- I fixed the JNI part in the `levelZeroContext.cpp` file to perform the check that if the `pNext` field is `-1`, then do not pass the value from Java to the descriptor.
- I took care of the corresponding cast checks for `_WIN32` and `LINUX` systems. I followed the conditions that pre-existed.

**To test on Linux OS:**
```bash
./scripts/copies.sh
./scripts/events.sh
./scripts/fences.sh
./scripts/kernelTimers.sh
./scripts/transfersTimers.sh
./scripts/largeBuffers.sh
./scripts/testZelloWorld.sh
```

**To test on Windows OS:**
```bash
.\scripts\copies.cmd
.\scripts\events.cmd
.\scripts\fences.cmd
.\scripts\kernelTimers.cmd
.\scripts\transfersTimers.cmd
.\scripts\largeBuffers.cmd
.\scripts\testZelloWorld.cmd
```

### CODE REVIEW

File: levelZeroLib/src/levelZeroContext.cpp [hunk 1]
```diff
@@ -328,7 +328,18 @@ JNIEXPORT jint JNICALL Java_uk_ac_manchester_tornado_drivers_spirv_levelzero_Lev
     env->SetIntField(javaCommandQueueDescriptor, field, commandQueueDesc.stype);
 
     field = env->GetFieldID(commanddescriptorClass, "pNext", "J");
-    env->SetLongField(javaCommandQueueDescriptor, field, reinterpret_cast<jlong>(commandQueueDesc.pNext));
+#ifdef _WIN32
+    jlong pNext = env->GetLongField(javaCommandQueueDescriptor, field);
+#else
+    jint pNext = env->GetLongField(javaCommandQueueDescriptor, field);
+#endif
```
Comment (jjfumero) [inline levelZeroLib/src/levelZeroContext.cpp:335]: Thanks @stratika . We can probably avoid duplicates with a C-Macro to get the right value for the pointer.

**Review by jjfumero — CHANGES_REQUESTED** [review]

File: levelZeroLib/src/levelZeroContext.cpp (hunk 1 above)
Comment (stratika) [reply levelZeroLib/src/levelZeroContext.cpp:335]: sure, I will try to fix that.

Comment (stratika) [reply levelZeroLib/src/levelZeroContext.cpp:331]: lust updated the file, let me know what you think.

File: levelZeroLib/src/ze_log.h [hunk 2]
```diff
@@ -55,4 +55,9 @@
         << std::endl;                                             \
     }
 
+#define ASSIGN_IF_NOT_MINUS_ONE(value, cpp_target, type) \
+    if (value != -1) { \
+        cpp_target = reinterpret_cast<type>(value); \
+    }
+
```
Comment (jjfumero) [inline levelZeroLib/src/ze_log.h:62]: Strange location for this Macro. I will open a new file for this type of macros.

File: levelZeroLib/src/ze_log.h [hunk 3]
```diff
@@ -55,4 +55,9 @@
         << std::endl;                                             \
     }
 
+#define ASSIGN_IF_NOT_MINUS_ONE(value, cpp_target, type) \
```
Comment (jjfumero) [inline levelZeroLib/src/ze_log.h:58]: Rename to `ASSIGN_POINTER_VALUE`

**Review by jjfumero — APPROVED** [review]

**Comment by stratika** [comment]
I tested the branch at this stage, on Windows OS. We are ready to merge.

**Pull request merged**